tools/generator/dfg/nrf/nrf_device_tree.py

```diff
@@ -0,0 +1,240 @@
+            if e.name in order:
+                if e.name in ['attribute-flash', 'attribute-eeprom', 'attribute-ram']:
+                    return (order.index(e.name), int(e['value']))
+                else:
+                    return (order.index(e.name), e['value'])
+            return (len(order), -1)
+        # tree.addSortKey(topLevelOrder)
+
+        # NRFDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-flash')
+        # NRFDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-ram')
+        # NRFDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-eeprom')
+        # NRFDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-mcu')
+
+        def driverOrder(e):
+            if e.name == 'driver':
+                if e['name'] == 'core':
+                    # place the core at the very beginning
+                    return ('aaaaaaa', e['type'])
+                if e['name'] == 'gpio':
+                    # place the gpio at the very end
+                    return ('zzzzzzz', e['type'])
+                # sort remaining drivers by type and compatible strings
+                return (e['name'], e['type'])
+            return ("", "")
+        tree.addSortKey(driverOrder)
+
+        # Core
+        core_child = tree.addChild('driver')
+        core_child.setAttributes('name', 'core', 'type', p['core'])
+        core_child.addSortKey(lambda e: (int(e['position']), e['name']) if e.name == 'vector' else (-1, ""))
+        core_child.addSortKey(lambda e: (e['name'], int(e['size'])) if e.name == 'memory' else ("", -1))
+
+        for section in p["memories"]:
+            memory_section = core_child.addChild("memory")
+            memory_section.setAttributes(["name", "access", "start", "size"], section)
+        # sort the node children by start address and size
+        core_child.addSortKey(lambda e: (int(e["start"], 16), int(e["size"])) if e.name == "memory" else (-1, -1))
+
+        # for memory in ['flash', 'ram', 'lpram', 'eeprom']:
+        #     if memory not in p: continue;
+        #     memory_section = core_child.addChild('memory')
+        #     memory_section.setAttribute('name', memory)
+        #     memory_section.setAttribute('size', p[memory])
+
+        for vector in p['interrupts']:
+            if int(vector['position']) < 0: continue;
+            vector_section = core_child.addChild('vector')
+            vector_section.setAttributes(['position', 'name'], vector)
+
+        modules = {}
+        for m, i in p['modules']:
+            # filter out non-peripherals: fuses, micro-trace buffer
+            if m in ['fuses', 'mtb', 'systemcontrol', 'systick', 'hmatrixb', 'hmatrix']: continue;
+            if m not in modules:
+                modules[m] = [i]
+            else:
+                modules[m].append(i)
+
+        # add all other modules
+        for name, instances in modules.items():
+            driver = tree.addChild('driver')
+            dtype = name
+            compatible = 'nrf'
```

Comment:
You would use `{platform}{family}`, so `nrf51`, `nrf52`, so that only the driver type is enough to decide module availability. At least that's what we found useful in modm.